src/index/annoy/annoy.cc

```diff
@@ -139,23 +139,21 @@ class AnnoyIndexNode : public IndexNode {
 
         auto rows = dataset.GetRows();
         auto dim = dataset.GetDim();
-        auto p_ids = dataset.GetIds();
+        auto ids = dataset.GetIds();
 
-        float* p_x = nullptr;
+        float* data = new (std::nothrow) float[dim * rows];
```

Comment:
Why put `new` outside try block? If you use `std::nothrow`, it might be better to test for `nullptr` if it fails to allocate. Although it might throw an exception when accessing it, it is unclear where the exception comes from.

Reply:
updated